pvlib/pvsystem.py

```diff
@@ -1019,72 +1049,86 @@ def pvwatts_ac(self, pdc):
 
     @property
     @_unwrap_single_value
+    @_check_deprecated_passthrough
     def module_parameters(self):
         return tuple(array.module_parameters for array in self.arrays)
 
     @property
     @_unwrap_single_value
+    @_check_deprecated_passthrough
     def module(self):
         return tuple(array.module for array in self.arrays)
 
     @property
     @_unwrap_single_value
+    @_check_deprecated_passthrough
     def module_type(self):
         return tuple(array.module_type for array in self.arrays)
 
     @property
     @_unwrap_single_value
+    @_check_deprecated_passthrough
     def temperature_model_parameters(self):
         return tuple(array.temperature_model_parameters
                      for array in self.arrays)
 
     @temperature_model_parameters.setter
```

Comment:
The `setter`s were added in #1076, so the deprecation requirement is a little more subtle. I think we either need to implement deprecation setters for each attribute or implement a `__setattr__` like this from modelchain:

https://github.com/pvlib/pvlib-python/blob/7eae1fc58cd693708c65102294a3991e6fbc328a/pvlib/modelchain.py#L503-L510

For that matter we could implement a `__getattr__` instead of properties, like this one:

https://github.com/pvlib/pvlib-python/blob/7eae1fc58cd693708c65102294a3991e6fbc328a/pvlib/modelchain.py#L491-L501

I don't know that one approach is better than another.

Reply:
I'm not following the issue here, can you expand on what's different about the `setter`s?

Reply:
After trying to come up with an example, I'm realizing the problem is a little different than I first described.

There are currently setters for only 4 properties:

* temperature_model_parameters
* surface_tilt
* surface_azimuth
* racking_model

The remaining properties do not contain setters. At first I thought this meant that for all of the other properties, a user could set something on a `PVSystem` object that's different than what they'd get back from a getter that inspects the Arrays. But I'm now reminded that the user simply will not be able to set them.

```python
In [5]: system = PVSystem([Array()])

In [6]: system.albedo = 0.3
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-d89b04b86f95> in <module>
----> 1 system.albedo = 0.3

AttributeError: can't set attribute
```

So the issue is that the user can set some but not all of the deprecated attributes. I'd prefer if they could set none or all.

I brought up the `__getattr__` and `__setattr__` methods only as an alternative to the properties and setters.

Reply:
> So the issue is that the user can set some but not all of the deprecated attributes. I'd prefer if they could set none or all.

I see, and agree it should be consistent.  I think omitting the setters was a (probably unintentional?) breaking change, so I think being able to set all (with a deprecation warning) makes sense.  I've added setters where they were missing.